### PR TITLE
API Assignment-1

### PR DESCRIPTION
**Pull request by waghmaresuyog**

### CODE REVIEW

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public Items(String productId, String productName, String productPrice, String productBrand) {
+        this.productId = productId;
+        this.productName = productName;
+        this.productPrice = productPrice;
+        this.productBrand = productBrand;
+    }
+
+    private static final Logger log = Logger.getLogger("Product.class");
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: It should be current className.class

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    @BeforeTest
+    public void getLoggerDisplay() {
+        PropertyConfigurator.configure("log4j2.properties");
+    }
+
+    @Test(priority = 1)
+    public void checkStatusCode() {
+        {
+            baseURI = ReusableMethod.getUrl();
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: Not using baseURI anywhere

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Test(priority = 3)
+    public void checkContent() {
+        Response response = RestAssured.get(url).then().extract().response();
+        JsonPath jsonObject = new JsonPath(response.asString());
+        List<Items> productList = jsonObject.getList("products");
+        List<Items> mockData = new ArrayList<>();
+        Items productLists = new Items("1", "Blue Top", "Rs. 500", "Polo");
+        mockData.add(productLists);
+        productLists = new Items("2", "Men Tshirt", "Rs. 400", "H&M");
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: Use getter and setter to get and set the values

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Test(priority = 4)
+    public void checkLength() {
+        var response = given().when().get(url).then().extract().asString();
+        JsonPath jsonResponse = new JsonPath(response);
+        var idLength = jsonResponse.getInt("products.id.size()");
+        log.info("The length is : " + idLength + " products");
+        Assert.assertEquals(idLength, 34, "number of products are Not expected");
+    }
+}
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java:89]: add readme file for problem statement and results

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/APIAssignments/Items_ReadMe [hunk 5]
```diff
@@ -0,0 +1,375 @@
...agsoup-1.2.1.jar;C:\Users\suyogw\.m2\repository\io\rest-assured\json-path\4.5.1\json-path-4.5.1.jar;C:\Users\suyogw\.m2\repository\org\codehaus\groovy\groovy-json\3.0.9\groovy-json-3.0.9.jar;C:\Users\suyogw\.m2\repository\io\rest-assured\rest-assured-common\4.5.1\rest-assured-common-4.5.1.jar;C:\Users\suyogw\.m2\repository\io\rest-assured\xml-path\4.5.1\xml-path-4.5.1.jar;C:\Users\suyogw\.m2\repository\org\apache\commons\commons-lang3\3.11\commons-lang3-3.11.jar;C:\Users\suyogw\.m2\repository\org\testng\testng\7.6.1\testng-7.6.1.jar;C:\Users\suyogw\.m2\repository\com\google\code\findbugs\jsr305\3.0.2\jsr305-3.0.2.jar;C:\Users\suyogw\.m2\repository\org\slf4j\slf4j-api\1.7.36\slf4j-api-1.7.36.jar;C:\Users\suyogw\.m2\repository\com\beust\jcommander\1.82\jcommander-1.82.jar;C:\Users\suyogw\.m2\repository\org\webjars\jquery\3.6.0\jquery-3.6.0.jar;C:\Users\suyogw\.m2\repository\log4j\log4j\1.2.12\log4j-1.2.12.jar;C:\Users\suyogw\.m2\repository\org\json\json\20230227\json-20230227.jar" com.intellij.rt.testng.RemoteTestNGStarter -usedefaultlisteners false -socket65345 @w@C:\Users\suyogw\AppData\Local\Temp\idea_working_dirs_testng.tmp -temp C:\Users\suyogw\AppData\Local\Temp\idea_testng.tmp
```
Comment (amitsahu8) [inline API_Auto/src/test/java/APIAssignments/Items_ReadMe]: remove logs

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 6]
```diff
@@ -58,32 +47,47 @@ public void checkProductList() {
...
+        var productPrice = reusableMethod.price;
+        var productBand = reusableMethod.brand;
+
+        Product objectProduct = new Product();
+        objectProduct.setProductId(String.valueOf(productID));
+        objectProduct.setProductName(productName);
+        objectProduct.setProductPrice(productPrice);
+        objectProduct.setProductBrand(productBand);
+        objectProduct.getProductId();
+        log.info(objectProduct.getProductId());
```
Comment (amitsahu8) [inline API_Auto/src/test/java/APIAssignments/Items.java]: remove unnecessary info log

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/APIAssignments/Product.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+public class Product {
+
+    public Product() {
+    }
+
+    String productId;
+    String productName;
+    String productPrice;
+    String productBrand;
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Product.java]: use private here

File: API_Auto/src/test/java/APIAssignments/ReusableMethod.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+package APIAssignments;
+
+public class ReusableMethod {
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/ReusableMethod.java]: class name should be proper

File: API_Auto/src/test/java/APIAssignments/ReusableMethod.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+    public static String getUrl() {
+        String url = "https://automationexercise.com/api/productsList";
+        return url;
+    }
+
+    int productId = 1;
+    String productName = "Blue Top";
+    String price = "Rs. 500";
+    String brand = "Polo";
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/ReusableMethod.java]: Use get setters to assign the value

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 1 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 3 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 4 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java:89]: done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 6 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+public class Items {
+
+    private static String url;
+
+    public Items() {
+        url = ReusableMethod.getUrl();
+    }
+
+    private static final Logger log = Logger.getLogger("Items.class");
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: Use Logger from Log4j library

File: API_Auto/src/test/java/APIAssignments/ReusableMethod.java (hunk 9 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/ReusableMethod.java]: done

File: API_Auto/src/test/java/APIAssignments/ReusableMethod.java (hunk 8 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/ReusableMethod.java]: done

File: API_Auto/src/test/java/APIAssignments/Product.java (hunk 7 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Product.java]: done

File: API_Auto/src/test/java/APIAssignments/Items_ReadMe (hunk 5 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items_ReadMe]: done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 2 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        JsonPath jsonObject = new JsonPath(response.asString());//response convert json into string
+        List<Product> productsArray = jsonObject.get("products");
+        List<Product> getProductData = new ArrayList<>();
+
+        Product objectProduct = new Product();
+        objectProduct.setProductId(String.valueOf(1));
+        objectProduct.setProductName("Blue Top");
+        objectProduct.setProductPrice("Rs. 500");
+        objectProduct.setProductBrand("Polo");
+        objectProduct.getProductId();
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: No use of this

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 11 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: Done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 10 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: Done

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+public class Items {
+
+    private static String url;
+
+    public Items() {
+        url = BaseUrl.getUrl();
+    }
+
+    private static Logger log = LogManager.getLogger("Items.class");
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: use final here

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+    @BeforeTest
+
+    public void getLoggerDisplay() {
+        PropertyConfigurator.configure("log4j2.properties");
+    }
+
+
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: remove extra space

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+    private static String url;
+
+    public Items() {
+        url = BaseUrl.getUrl();
+    }
+
+    private static Logger log = LogManager.getLogger("Items.class");
+
+    @BeforeTest
+
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/Items.java]: remove extra space

File: API_Auto/src/test/java/APIAssignments/BaseUrl.java [hunk 15]
```diff
@@ -0,0 +1,10 @@
+package APIAssignments;
+
+public class BaseUrl {
+
+    public static String getUrl() {
+        String url = "https://automationexercise.com/api/productsList";
+        return url;
```
Comment (prashantkawa) [inline API_Auto/src/test/java/APIAssignments/BaseUrl.java]: Please try using .properties file for URL

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/APIAssignments/BaseUrl.java (hunk 15 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/BaseUrl.java]: Done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 14 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 13 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: Done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 12 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: Done

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+    @Test(priority = 2)
+    public void checkProductList() {
+        var getList = given().when().get(url).then().log().all().toString();
+        log.info("Product List is " + getList);
+    }
+
+    @Test(priority = 3)
+    public void checkContent() {
+
```
Comment (vrushalee123) [inline API_Auto/src/test/java/APIAssignments/Items.java]: Remove extra space.

File: API_Auto/src/test/java/APIAssignments/Items.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    }
+    private static final Logger log = LogManager.getLogger("Items.class");
+    @BeforeTest
+    public void getLoggerDisplay() {
+        PropertyConfigurator.configure("log4j2.properties");
+    }
+    @Test(priority = 1)
+    public void checkStatusCode() {
+        {
```
Comment (vrushalee123) [inline API_Auto/src/test/java/APIAssignments/Items.java]: Why two curly braces needed.

**Review by vrushalee123 — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 17 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

File: API_Auto/src/test/java/APIAssignments/Items.java (hunk 16 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Items.java]: done

**Review by vrushalee123 — APPROVED** [review]

**Pull request merged**